game-core/src/main/java/games/strategy/triplea/ui/UserActionText.java

```diff
@@ -75,6 +81,14 @@ public String getNotificationSuccessOthers(final String actionKey) {
     return getMessage(actionKey, OTHER_NOTIFICATION_SUCCESS);
   }
 
+  public String getNotificationSuccessTarget(final String actionKey) {
+    if (hasMessage(actionKey, TARGET_NOTIFICATION_SUCCESS)) {
+      return getMessage(actionKey, TARGET_NOTIFICATION_SUCCESS);
+    } else {
+      return getNotificationSuccessOthers(actionKey);
```

Comment:
Not following this logic. Shouldn't this return "NONE"?

Reply:
This is so maps that implement OTHER notifications continue to work as-is. That is, if TARGET is not specified then OTHER is sent to all other players, as is now.

There are other ways to do this, such as returning NONE and having logic in the caller to check for that and in the case of NONE, send OTHER to all players. This way is actually a bit cleaner since the logic the caller doesn't need to have complicated logic to handle it.

But this way is actually more flexible - for example if mapmaker wanted to only send notifications to players and others, but not target - they can do it with this set up (by setting TARGET to NONE). If we instead did it the other way, then this wouldn't be possible.

Reply:
Ok, I now see what you are doing here. I'm fine with that as I was just thinking about it the first way you describe.